crypto/src/encryption/mod.rs

```diff
@@ -0,0 +1,293 @@
+//! A suite of Authenticated Encryption with Associated Data (AEAD) cryptographic ciphers.
+//!
+//! Each AEAD algorithm provides [`SymmetricEncryptor::encrypt_easy`] and [`SymmetricEncryptor::decrypt_easy`] methods which hides the complexity
+//! of generating a secure nonce of appropriate size with the ciphertext.
+//! The [`SymmetricEncryptor::encrypt_easy`] prepends the nonce to the front of the ciphertext and [`SymmetricEncryptor::decrypt_easy`] expects
+//! the nonce to be prepended to the front of the ciphertext.
+//!
+//! More advanced users may use [`SymmetricEncryptor::encrypt`] and [`SymmetricEncryptor::decrypt`] directly. These two methods require the
+//! caller to supply a nonce with sufficient entropy and should never be reused when encrypting
+//! with the same `key`.
+//!
+//! The convenience struct [`SymmetricEncryptor`] exists to allow users to easily switch between
+//! algorithms by using any algorithm that implements the [`Encryptor`] trait.
+//!
+//! [`ChaCha20Poly1305`] is the only algorithm currently supported,
+//! as it is the only one used by the iroha p2p transport protocol.
+
+mod chacha20poly1305;
+
+use std::io::{Read, Write};
+
+use aead::{
+    generic_array::{typenum::Unsigned, ArrayLength, GenericArray},
+    Aead, Error, KeyInit, Payload,
+};
+use rand::{rngs::OsRng, RngCore};
+
+pub use self::chacha20poly1305::ChaCha20Poly1305;
```

Comment:
```suggestion
pub use chacha20poly1305::ChaCha20Poly1305;
```

Reply:
This is required for disambiguation, we use a dependency named `chacha20poly1305`